### ISSUE TITLE
xml diffwith whitespaces 

### ISSUE THREAD

**Issue by dmytro-dovhan** [xmlunit/xmlunit.net#36]
Faced that DiffBuilder ignores whitespaces between xml nodes and returns 'identical' in next example.
In case whitespace in 'uuid' value everything works as expected.  I assume behaviour like Java https://stackoverflow.com/questions/5721290/does-xmlunit-have-an-assert-to-ignore-whitespace

`var myControlXML = "<msg><uuid>0x00435A8C</uuid></msg>";`
`var myTestXML = "<msg><uuid>0x00435A8C</uuid>        </msg>";`
`var md = DiffBuilder.Compare(myControlXML).WithTest(myTestXML).CheckForIdentical().Build();`

What am I doing wrong? I tried different versions starteed with 2.0.0

**Comment by bodewig** [comment]
The Stackoverflow article you link to is about XMLUnit 1.x, 2.x (both Java and .NET) has a completely different API.

Your test XML contains what XML calls "element content whitespace", you can get rid of the difference by telling XMLUnit to ignore it, see https://github.com/xmlunit/user-guide/wiki/Providing-Input-to-XMLUnit#elementcontentwhitespacestrippedsource . You could either wrap your input manually or - as you are using `DiffBuilder` calling `IgnoreWhitespace` will do.

```
DiffBuilder.Compare(myControlXML).WithTest(myTestXML).IgnoreWhitespace().CheckForIdentical().Build()
```

see https://github.com/xmlunit/user-guide/wiki/DiffBuilder

**Comment by bodewig** [comment]
Just saw the second answer at Stackoverflow recommended the same thing I did, you probably should upvote it.

**Comment by bodewig** [comment]
and `IgnoreElementContentWhitespace` should be enough in your case. Should have recommended that from the beginning.

**Comment by dmytro-dovhan** [comment]
![image](https://user-images.githubusercontent.com/16334310/102463367-b0f15000-4053-11eb-8822-133a591cfe12.png)
sorry, I can't understand what I'm doing wrong :)

**Comment by bodewig** [comment]
I'm sorry, but I have no idea what you are asking for. Does the result of diffing the two XMLs return any differences?

**Comment by dmytro-dovhan** [comment]
It seems now I realized that I want something imposible. I need to check formating during comparing files in my app. I want to use AttributeFilter/NodeMatcher and the same time check xml formating. Now I understand that engine have to parse whole xml and store it in XmlNodes structure to allow all provided manipulations. But I compared whitespeces outside xml value.

I have to add some logic to check formating. I'll just serialize/deserialize my test xml file and compare it as strings.

Sorry to create this issue while didn't sort out with docs and my requirements. Thanks for great lib and really fast responsens for my stupid issue.
